Революция или эволюция? (часть 1)

Sasha20101989 opened the pull request:
Так и не понял по заданию требовалось ли отобразить форму редактирования точки или рандомную

---
:mortar_board: [Революция или эволюция? (часть 1)](https://up.htmlacademy.ru/ecmascript/19/user/2033003/tasks/7)

Sasha20101989 committed:
реализовал замену на форму редактирования

Так и не понял по заданию требовалось ли отобразить форму редактирования точки или рандомную

k0t-pirat commented on src/presenter/content-presenter.js:
```diff
@@ -1,28 +1,61 @@
   }
 
+  #renderPoint(point, randomeDestination, allDestinations){
+    const pointComponent = new WaypointView({waypoint: point, destination: randomeDestination, allDestinations: allDestinations});
```
Я бы рекомендовал все манипуляции с данными проводить в модели, и пусть модель отдаёт одну структуру данных с готовыми точками, а не три "сырых" структуры данных, так мы избавимся от необходимости всюду их передавать.

Sasha20101989 replied:
question

k0t-pirat commented on src/model/waypoint-model.js:
```diff
@@ -6,7 +6,7 @@ export default class WaypointModel{
     return this.#waypoints.sort((a, b) => a.dateFrom > b.dateFrom ? 1 : -1);
```
здесь можно написать проще waypoints.sort((a, b) => a.dateFrom - b.dateFrom)

Sasha20101989 replied:
done

k0t-pirat commented on src/model/destination-model.js:
```diff
@@ -8,7 +8,7 @@ export default class DestinationModel{
   }
 
-  getDestinations(){
+  get destinations(){
```
мне кажется, лучше одну можель использовать, а не 3 разных

Sasha20101989 replied:
question

k0t-pirat commented on src/presenter/content-presenter.js:
```diff
@@ -1,28 +1,61 @@
+    editPointComponent.element.querySelector('form').addEventListener('reset', () => {
+    });
+    editPointComponent.element.querySelector('form').addEventListener('submit', (evt) => {
+      evt.preventDefault();
```
нет действия replaceEditToPoint по submit

Sasha20101989 replied:
done

Sasha20101989 replied:
и удаление подписки на esc

k0t-pirat commented:
Технически задание выполнено верно, но логически некорректно, так как у нас должно отображаться редактирование той точки, которую мы выбрали при нажатии на стелочку, сейчас же данные точки просмотра и редактирования не совпадают. Скорее всего эта проблема тянется из прошлого задания, так как компонент EditPointView принимает часть данных из моков, а не из модели. С моками на данном этапе работает только модель, презентер и Вью ничего о них знать не должны. При прошлой проверке я этот момент упустил, но сегодня поправим.

Sasha20101989 committed:
исправил, часть ошибок

в том числе удалил не нужную передачу destination во вью точки

Sasha20101989 committed:
Удалил не нужную модель и изменил вьюхи

Sasha20101989 committed:
разметка не очень большая, решил не выделять в отдельную функцию

Sasha20101989 committed:
исправил ошибки проверки lint

Sasha20101989 committed:
привел дату к стилю требуемому в ТЗ

Sasha20101989 committed:
однотипные констаны собрал в перечисления

k0t-pirat commented on src/model/waypoint-model.js:
```diff
@@ -1,12 +1,68 @@
+    return this.#offers;
   }
+
+  get humanazeWaypoints(){
```
очень мелкая ошибка - humanize, а не humanaze, а так же так как это геттер, а не функция, то лучше использовать существительное, а не глагол, например, humanizedWaypoints

Sasha20101989 replied:
done

k0t-pirat commented on src/model/waypoint-model.js:
```diff
@@ -1,12 +1,68 @@
+  dateTo: point.dateTo,
+  destination: destination,
+  id: point.id,
+  isFavorite: returnRandomBool(),
```
значение isFavorite тоже пусть из моков приходит

Sasha20101989 replied:
тут я не доглядел, тут будет правильно point.IsFavorite

done

k0t-pirat reviewed (commented):
![image](https://user-images.githubusercontent.com/30899837/209580401-008e5247-3b8c-4c86-a59b-66ec856b368d.png)
в editPointView не исправлен приём данных, нужно сделать по аналогии с waypointView

k0t-pirat commented on src/model/waypoint-model.js:
```diff
@@ -1,12 +1,68 @@
   }
+
+  get humanazeWaypoints(){
+    const cloneWaypoints = Object.assign([], this.#waypoints);
```
Object.assign применяется для обычных объектов, а не для массивов, для массивов стоит применить .slice(), либо клонирование с поиощью rest-оператора - newArray = [...oldArray]

Sasha20101989 replied:
Урок усвоил)
done

Sasha20101989 committed:
исправил ошибки именования и получения данных

Sasha20101989 committed:
Добавил все типы и все оферы которые относятся к типу точки, в отдельные массивы точки

Sasha20101989 committed:
удалил не используемую функцию

Sasha20101989 committed:
Поправил стилизацию и изменил итерации по структуре данных

Sasha20101989 committed:
переписал функцию проверки выбранного input

понимаю что можно как то еще проще сделать, подумаю позже как упростить, добавил себе в задачи
